### PR TITLE
EDU-155 Replace superagent with axios

### PR DESCRIPTION
**Pull request by adrianaluchian**
Closes https://educandu.atlassian.net/browse/EDU-155

Due to the fact that this ticket required multiple possibly breaking changes, I would suggest we hold off on merging until the Import epic is on ELMU and OMA staging.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -135,8 +135,11 @@ function runJest(...flags) {
 }
 
 const downloadCountryList = async lang => {
-  const res = await superagent.get(`https://raw.githubusercontent.com/umpirsky/country-list/master/data/${lang}/country.json`);
-  await fs.writeFile(`./src/data/country-names/${lang}.json`, JSON.stringify(JSON.parse(res.text), null, 2), 'utf8');
+  const res = await axios.get(
+    `https://raw.githubusercontent.com/umpirsky/country-list/master/data/${lang}/country.json`,
```
Comment (ahelmberger) [inline gulpfile.js:139]: I know that it was that sloppy before but with this change we would have the chance to correct all URL interpolations with proper escaping like `https://raw.githubusercontent.com/umpirsky/country-list/master/data/${encodeURIComponent(lang)}/country.json`. I'll put a '+1' on each place I see and leave the decision to you (in a basically all cases it's more a theoretical matter as we control the params there)

File: src/services/cdn-api-client.js [hunk 2]
```diff
@@ -9,23 +9,31 @@ class CdnApiClient {
 
   getObjects(prefix) {
     return this.httpClient
-      .get(`/api/v1/cdn/objects?prefix=${prefix}`)
-      .accept('json')
-      .then(res => res.body);
+      .get(
+        `/api/v1/cdn/objects?prefix=${prefix}`,
```
Comment (ahelmberger) [inline src/services/cdn-api-client.js:13]: +1

File: src/services/document-api-client.js [hunk 3]
```diff
@@ -9,73 +9,90 @@ class DocumentApiClient {
...
+      .then(res => res.data);
   }
 
   getRevisionTagSuggestions(tagsSuggestionQuery) {
     return this.httpClient
-      .get(`/api/v1/docs/revisions/tags/${tagsSuggestionQuery}`)
-      .accept('json')
-      .then(res => res.body);
+      .get(
+        `/api/v1/docs/revisions/tags/${tagsSuggestionQuery}`,
```
Comment (ahelmberger) [inline src/services/document-api-client.js:63]: +1

File: src/services/document-api-client.js [hunk 4]
```diff
@@ -9,73 +9,90 @@ class DocumentApiClient {
...
+      .then(res => res.data);
   }
 
   getDocumentTagSuggestions(tagsSuggestionQuery) {
     return this.httpClient
-      .get(`/api/v1/docs/tags/${tagsSuggestionQuery}`)
-      .accept('json')
-      .then(res => res.body);
+      .get(
+        `/api/v1/docs/tags/${tagsSuggestionQuery}`,
```
Comment (ahelmberger) [inline src/services/document-api-client.js:72]: +1

File: src/services/document-api-client.js [hunk 5]
```diff
@@ -9,73 +9,90 @@ class DocumentApiClient {
...
   }
 
   archiveDocument(documentKey) {
     return this.httpClient
-      .patch(`/api/v1/docs/${documentKey}/archive`)
-      .accept('json')
-      .send()
-      .then(res => res.body);
+      .patch(
+        `/api/v1/docs/${documentKey}/archive`,
```
Comment (ahelmberger) [inline src/services/document-api-client.js:81]: +1

File: src/services/document-api-client.js [hunk 6]
```diff
@@ -9,73 +9,90 @@ class DocumentApiClient {
...
   }
 
   unarchiveDocument(documentKey) {
     return this.httpClient
-      .patch(`/api/v1/docs/${documentKey}/unarchive`)
-      .accept('json')
-      .send()
-      .then(res => res.body);
+      .patch(
+        `/api/v1/docs/${documentKey}/unarchive`,
```
Comment (ahelmberger) [inline src/services/document-api-client.js:91]: +1

File: src/services/user-api-client.js [hunk 7]
```diff
@@ -9,81 +9,91 @@ class UserApiClient {
...
 
   saveUserRoles({ userId, roles }) {
     return this.httpClient
-      .post(`/api/v1/users/${userId}/roles`)
-      .type('json')
-      .accept('json')
-      .send({ roles })
-      .then(res => res.body);
+      .post(
+        `/api/v1/users/${userId}/roles`,
```
Comment (ahelmberger) [inline src/services/user-api-client.js:62]: +1

File: src/services/user-api-client.js [hunk 8]
```diff
@@ -9,81 +9,91 @@ class UserApiClient {
...
 
   saveUserLockedOutState({ userId, lockedOut }) {
     return this.httpClient
-      .post(`/api/v1/users/${userId}/lockedOut`)
-      .type('json')
-      .accept('json')
-      .send({ lockedOut })
-      .then(res => res.body);
+      .post(
+        `/api/v1/users/${userId}/lockedOut`,
```
Comment (ahelmberger) [inline src/services/user-api-client.js:92]: +1

**Review by ahelmberger — APPROVED** [review]

File: gulpfile.js (hunk 1 above)
Comment (adrianaluchian) [reply gulpfile.js:139]: Yeah very good point, let's do that then!
Thanks for the +1 to help find them, will change.

Comment (adrianaluchian) [reply gulpfile.js]: -1

File: src/services/cdn-api-client.js (hunk 2 above)
Comment (adrianaluchian) [reply src/services/cdn-api-client.js]: -1

File: src/services/document-api-client.js (hunk 3 above)
Comment (adrianaluchian) [reply src/services/document-api-client.js]: -1

File: src/services/document-api-client.js (hunk 4 above)
Comment (adrianaluchian) [reply src/services/document-api-client.js]: -1

File: src/services/document-api-client.js (hunk 5 above)
Comment (adrianaluchian) [reply src/services/document-api-client.js]: -1

File: src/services/document-api-client.js (hunk 6 above)
Comment (adrianaluchian) [reply src/services/document-api-client.js]: -1

File: src/services/user-api-client.js (hunk 7 above)
Comment (adrianaluchian) [reply src/services/user-api-client.js]: -1

File: src/services/user-api-client.js (hunk 8 above)
Comment (adrianaluchian) [reply src/services/user-api-client.js]: -1

**Review by laurentiu-ilici — APPROVED** [review]

**Pull request merged**